### PR TITLE
chore: update dependencies (#90314)

### PR DESCRIPTION
**Pull request by cortlepp-intershop**
# Changes

- update gradle wrapper to 8.5
- update dependencies
- update Java version to 17 LTS
- use the same build/release configuration for all gradle plugins from [the new .github repo](https://github.com/IntershopCommunicationsAG/.github)

### CODE REVIEW

**Review by m-raab — APPROVED** [review]

**Review by DevDavido — APPROVED** [review]

**Review by skiesewetter-intershop — APPROVED** [review]

**Comment by cortlepp-intershop** [comment]
Do I need to update the version in `build.gradle.kts` before merging this PR? And if so, is this a major change? So `5.0.0`?

**Comment by DevDavido** [comment]
Since this PR drops support for < Java 17, it's a breaking change as the published artifacts are compiled for at least JRE version 17. Therefore, please publish a new major version `5.0.0` then.

The version string needs to be adapted in the `build.gradle.kts` and in the `README.asciidoc` (see `latestRevision`). Also please revise the documentation regarding the supported Gradle version.

File: build.gradle.kts [hunk 1]
```diff
@@ -64,41 +60,38 @@ tasks {
...
         testLogging.showStandardStreams = true
 
-        this.javaLauncher.set( project.javaToolchains.launcherFor {
-            languageVersion.set(JavaLanguageVersion.of(11))
-        })
-
         // Gradle versions for test
-        systemProperty("intershop.gradle.versions", "7.2,7.5.1")
-        systemProperty("intershop.test.base.dir", (File(project.buildDir, "test-working")).absolutePath)
+        systemProperty("intershop.gradle.versions", "8.4,8.5")
```
Comment (DevDavido) [inline build.gradle.kts:64]: May update the README too regarding the Gradle version(s)

**Review by DevDavido — CHANGES_REQUESTED** [review]

File: build.gradle.kts [hunk 2]
```diff
@@ -201,17 +197,15 @@ repositories {
...
     }
     api("org.spockframework:spock-junit4")
-    api("commons-io:commons-io:2.11.0")
-    api("com.sun.xml.bind:jaxb-impl:4.0.0")
-
-    implementation("jakarta.xml.bind:jakarta.xml.bind-api:4.0.0")
+    api("commons-io:commons-io:2.14.0")
+    api("com.sun.xml.bind:jaxb-impl:4.0.3")
+    implementation("jakarta.xml.bind:jakarta.xml.bind-api:4.0.1")
     implementation("junit:junit:4.13.2")
```
[the remark this answers is not in the document]
Comment (skiesewetter-intershop) [reply build.gradle.kts:208]: Spock only supports JUnit4 as far as I know.

**Review by DevDavido — APPROVED** [review]

**Pull request merged**